src/Session.cc

```diff
@@ -0,0 +1,156 @@
+#include "Session.h"
+#include "guileSessionManagerDeff.h"
+
+namespace opencogservices
+{
+Session::Session(int write_fd, int read_fd)
+{
+	this->mPid = -1;
+	this->mWriteFd = write_fd;
+	this->mReadFd = read_fd;
+}
+
+Session::Session(int pid, int write_fd, int read_fd)
+{
+	this->mPid = pid;
+	this->mWriteFd = write_fd;
+	this->mReadFd = read_fd;
+}
+
+Session::Session()
+{
+	this->mPid = -1;
+	this->mWriteFd = -1;
+	this->mReadFd = -1;
+}
+
+Session::~Session()
+{
+}
+
+int Session::sendMsg(const string &msg)
+{
+	string s_msg = msg;
+	s_msg.push_back(SSM_COMMAND_DELIMITER);
+	int status = write(mWriteFd, s_msg.c_str(), s_msg.length());
+
+	if (status != 0) {
+		return S_OP_ERROR_CAN_NOT_SEND_MSG;
+	}
+
+	return S_OP_MESSAGE_SENT;
+}
+
+int Session::sendMsg(const char *msg)
```

Comment:
Remove code replication. This method should call the previous one passing the std::string